File: education_brain/knowledge/models/question.py
"""题库领域数据模型 — 对应 PLAN.md §5.2"""

from pydantic import BaseModel, Field

class QuestionBank(BaseModel):
    """题库 — 来自 题目资料.md 的 ## 级别
    
    示例:
    ## 通用程序设计题库

    - 题库编码: general_purpose_programming_bank
    """
    bank_code: str
    bank_name: str
    domain_tags: list[str] = Field(default_factory=list)
    question_count: int = 0

class QuestionOption(BaseModel):
    """选项"""
    label: str
    content: str

class QuestionItem(BaseModel):
    """题目 — 来自 ### question_code 级别
    示例:
    ### general_purpose_programming_bank_q002

    - **题型**: 单选题
    - **题干**: 关于数据类型转换，下面哪种做法最稳妥？
    - **选项**:
    A. 直接把任意文本当作数字参与运算
    B. 在转换前先校验输入格式，再执行显式转换
    C. 只要能运行，所有转换都可以忽略异常
    D. 所有语言都会自动处理错误格式
    - **答案**: B
    - **解析**: 显式校验后再转换，可以减少运行时错误并让问题更容易定位。
    """
    question_code: str
    bank_code: str
    question_type: str = ""
    stem: str = ""
    options: list[QuestionOption] = Field(default_factory=list)
    answer_key: str = ""
    reference_answer: str = ""
    analysis: str = ""
    raw_block: str = ""
    quality_flags: list[str] = Field(default_factory=list)

File: education_brain_front/src/app/mock/question.ts
import type { SearchQuestionsResponse } from '../types'

export function mockGetQuestions(_params: { keyword?: string; bank_code?: string; question_type?: string; page?: number; page_size?: number }): SearchQuestionsResponse {
  return {
    items: [
      {
        question_code: 'python_bank_q001', bank_code: 'python_bank', bank_name: 'Python 题库',
        question_type: 'single_choice', stem: 'Python 中用于定义函数的关键字是什么？',
        options: [{ label: 'A', content: 'func' }, { label: 'B', content: 'def' }, { label: 'C', content: 'function' }, { label: 'D', content: 'lambda' }],
        answer_key: ['B'], reference_answer: null, analysis: 'Python 使用 def 定义函数。', quality_flags: [],
      },
      {
        question_code: 'python_bank_q002', bank_code: 'python_bank', bank_name: 'Python 题库',
        question_type: 'multiple_choice', stem: '以下哪些是 Python 内置数据类型？',
        options: [{ label: 'A', content: 'list' }, { label: 'B', content: 'dict' }, { label: 'C', content: 'array' }, { label: 'D', content: 'tuple' }],
        answer_key: ['A', 'B', 'D'], reference_answer: null, analysis: 'array 不是 Python 内置类型，需要 import array 模块。', quality_flags: [],
      },
      {
        question_code: 'mysql_bank_q001', bank_code: 'mysql_bank', bank_name: 'MySQL 题库',
        question_type: 'single_choice', stem: 'SQL 中用于查询数据的关键字是？',
        options: [{ label: 'A', content: 'GET' }, { label: 'B', content: 'FETCH' }, { label: 'C', content: 'SELECT' }, { label: 'D', content: 'FIND' }],
        answer_key: ['C'], reference_answer: null, analysis: 'SELECT 是 SQL 标准查询语句。', quality_flags: [],
      },
      {
        question_code: 'python_bank_q003', bank_code: 'python_bank', bank_name: 'Python 题库',
        question_type: 'fill_blank', stem: 'Python 中使用 ______ 关键字导入模块。',
        options: [], answer_key: ['import'], reference_answer: 'import', analysis: 'Python 使用 import 关键字导入模块。',
        quality_flags: ['stem_too_short'],
      },
      {
        question_code: 'mysql_bank_q002', bank_code: 'mysql_bank', bank_name: 'MySQL 题库',
        question_type: 'true_false', stem: 'MySQL 中 PRIMARY KEY 允许 NULL 值。',
        options: [{ label: 'A', content: '正确' }, { label: 'B', content: '错误' }],
        answer_key: ['B'], reference_answer: null, analysis: 'PRIMARY KEY 不允许 NULL 值。', quality_flags: [],
      },
    ],
    pagination: { page: 1, page_size: 20, total: 5 },
  }
}
